test/common_test/collisions.cc

```diff
@@ -141,6 +147,133 @@ TYPED_TEST(CollisionTest, MeshAndPlane)
+    if (this->PhysicsEngineName(name) != "bullet-featherstone")
+      continue;
+    std::cout << "Testing plugin: " << name << std::endl;
+    gz::plugin::PluginPtr plugin = this->loader.Instantiate(name);
+
+    sdf::Root rootWorld;
+    const sdf::Errors errorsWorld =
+        rootWorld.Load(common_test::worlds::kGroundSdf);
+    ASSERT_TRUE(errorsWorld.empty()) << errorsWorld.front();
+
+    auto engine =
+        gz::physics::RequestEngine3d<CollisionMeshFeaturesList>::From(plugin);
+    ASSERT_NE(nullptr, engine);
+
+    auto world = engine->ConstructWorld(*rootWorld.WorldByIndex(0));
+    ASSERT_NE(nullptr, world);
+
+    // load the mesh into mesh manager first to create a cache
+    // so the model can be constructed later - needed by bullet-featherstone
+    const std::string meshFilename = gz::physics::test::resources::kChassisDae;
+    auto &meshManager = *gz::common::MeshManager::Instance();
+    ASSERT_NE(nullptr, meshManager.Load(meshFilename));
+
+    std::unordered_set<std::string> optimizations;
+    optimizations.insert("");
+    optimizations.insert("convex_decomposition");
+    optimizations.insert("convex_hull");
+
+    gz::math::Pose3d initialModelPose(0, 0, 2, 0, 0, 0);
+    // Test all optimization methods
+    for (const auto &optimizationStr : optimizations)
+    {
+      // create the chassis model
+      const std::string modelOptimizedName = "model_optimized_" + optimizationStr;
+      sdf::Root root;
+      sdf::Errors errors = root.LoadSdfString(getModelOptimizedStr(
+          optimizationStr, modelOptimizedName, initialModelPose));
+      ASSERT_TRUE(errors.empty()) << errors.front();
+      ASSERT_NE(nullptr, root.Model());
+      world->ConstructModel(*root.Model());
+
+      const std::string bodyName{"body"};
+      auto modelOptimized = world->GetModel(modelOptimizedName);
+      auto modelOptimizedBody = modelOptimized->GetLink(bodyName);
+
+      auto frameDataModelOptimizedBody =
+          modelOptimizedBody->FrameDataRelativeToWorld();
+
+      EXPECT_EQ(initialModelPose,
+                gz::math::eigen3::convert(frameDataModelOptimizedBody.pose));
+
+      // After a while, the mesh model should reach the ground and come to a stop
+      gz::physics::ForwardStep::Output output;
+      gz::physics::ForwardStep::State state;
+      gz::physics::ForwardStep::Input input;
+      std::size_t stepCount = 3000u;
+      for (unsigned int i = 0; i < stepCount; ++i)
+        world->Step(output, state, input);
+
+      frameDataModelOptimizedBody =
+          modelOptimizedBody->FrameDataRelativeToWorld();
+
+      // convex decomposition gives more accurate results
+      double tol = (optimizationStr == "convex_decomposition") ? 1e-3 : 1e-2;
+      EXPECT_NEAR(0.1,
+                  frameDataModelOptimizedBody.pose.translation().z(), tol);
+      EXPECT_NEAR(0.0, frameDataModelOptimizedBody.linearVelocity.z(), tol);
+
+      initialModelPose.Pos() += gz::math::Vector3d(0, 2, 0);
```

Comment:
In general, I thought testing with a V or U shaped object would really show the difference in behavior between `convex_hull` and `convex_decomposition`, but
I'm getting some unexpected penetration with `convex_decomposition`. I've attached the .sdf files.
![image](https://github.com/gazebosim/gz-physics/assets/206116/8abf5729-ddc9-4cf5-b011-2278d4b0364c)

Maybe we can use this in the test?

[test_conv_decomp.zip](https://github.com/gazebosim/gz-physics/files/14928199/test_conv_decomp.zip)

Reply:
I added an integration test for this. In the test I set `<max_convex_hulls>` to 64 which prevents this penetration. [2099054](https://github.com/gazebosim/gz-physics/pull/606/commits/2099054eef9a92bf473f595042954f1d99593af6)

The generated convex hulls looked fine with either 16 or 64 `max_convex_hulls` so something else is not right. This is something to look into next.